web/src/components/tooltip.js

```diff
@@ -24,42 +24,50 @@ const Tooltip = ({
 }) => {
   const isMobile = useSelector(state => state.application.isMobile);
 
-  const ref = useRef(null);
-  const width = useWidthObserver(ref);
-  const height = useHeightObserver(ref);
+  const {
+    ref,
+    width,
+    height,
+  } = useRefWidthHeightObserver();
 
   const screenWidth = window.innerWidth;
   const screenHeight = window.innerHeight;
 
   if (!position) return null;
 
+  // Note: at first render, width and height will be undefined
+  // They will only be set once the DOM node has been created
```

Comment:
Wait, according to `useRefWidthHeightObserver` below, they actually start with a value `0`. Maybe we should indeed change it to `null` or `undefined` or update this comment and put a more explicit condition below (e.g. `width > 0 && height > 0`) to avoid new corner cases and keep it simple.

Reply:
good point